### PR TITLE
Pufferfish, Tetrodotoxin, and Delicacies

### PR DESCRIPTION
**Pull request by colossusqw**
[LABEL][feature][catering][chemistry]

## About the PR 
This PR adds pufferfish as a uncommon aquarium fish spawn. Pufferfish requires gloves or chef training to properly handle, and will be poisonous unless prepared by a certified chef. Poorly prepared dishes will contain deadly amounts of tetrodotoxin, as well as risking poisoning the person preparing them, but correctly prepared ones will be safe for consumption. The poison that pufferfish contain is tetrodotoxin, a dangerous neurotoxin that can paralize and cause heart failure if not quickly flushed.
image

## Why's this needed? 
Someone suggested the idea on the forums, and there aren't any poisons exclusive to fishing, so pufferfish tetrodotoxin seemed like a perfect spot.  Tetrodotoxin was made with the intention of not being as immediately threatening and low dose deadly as other toxins, needing at least 15 units and 47 cycles(a minute and a half) to reach the heart effects, but have a quick surprise effect, with lots of heart related flavor effects along the way, which would lend itself well to more intrigue and RP focused poisonings. As for the fish, the idea of needing the chef for a "prestigious" set of meals that can be quite dangerous if not properly prepared seems to fit right in with the setting's wackyness.

## Changelog

```changelog
(u)Colossus
(*)Added pufferfish as a catchable saltwater aquarium fish, filled with tetrodotoxin, can be quite dangerous if prepared incorrectly.
(+)Tetrodotoxin remains dormant for 10 cycles, causes emotes, messages, slight bouts of misstep and stutters for 17 cycles, causes blurry vision, random oxygen deprivation and slowness, as well as loud heart palpitations for 20 cycles, and finally causes oxygen damage and cardiac arrest.
(+)Pufferfish come with lots of tetrodotoxin in them, and require someone with proper kitchen training(the chef) to separate the toxic liver from the fillet without either contaminating it or poisoning themselves.
```

### CODE REVIEW

**Comment by Gannets** [comment]
I haven't looked at the code but I like this idea a lot! exactly what I'd like from new fish

**Comment by mordent-goonstation** [comment]
Am I right in thinking that this is going to be a moderately silent killer when you stuff the fish into a reagent extractor to get the toxin out and then stick it in other foods?

**Comment by colossusqw** [comment]
> Am I right in thinking that this is going to be a moderately silent killer when you stuff the fish into a reagent extractor to get the toxin out and then stick it in other foods?

I could make it more obvious, but 18 cycles of random bouts of misstep and stutters followed by 15 cycles of very loud heartbeating and misc effects on a chem that will be stopped by any flusher doesn't seem super crazy to me. This and taking into account it being a fishing only poison.

File: code/modules/fishing/fish.dm [hunk 1]
```diff
@@ -285,6 +285,58 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/fish)
...
+			user.reagents.add_reagent("tetrodotoxin", 20)
+			src.reagents.remove_reagent("tetrodotoxin", 20)
+
+	proc/spikes_protected(mob/living/carbon/human/H)
+		if (H.hand)//gets active arm - left arm is 1, right arm is 0
+			if (istype(H.limbs.l_arm,/obj/item/parts/robot_parts) || istype(H.limbs.l_arm,/obj/item/parts/human_parts/arm/left/synth))
+				return TRUE
+		else
+			if (istype(H.limbs.r_arm,/obj/item/parts/robot_parts) || istype(H.limbs.r_arm,/obj/item/parts/human_parts/arm/right/synth))
+				return TRUE
```
Comment (mordent-goonstation) [inline code/modules/fishing/fish.dm:334]: Why would synthlimbs protect here?

**Review by mordent-goonstation — COMMENTED** [review]
Broadly, this thing has very little indication of *deadly poison* until cycle 30. So you have about 30 seconds to a) recognize that the escalating effects are what they are and deadly, b) handle the escalating effects in your quest for charcoal/to get to medbay/etc. Oh, and you're slowed for the entirety of that last band.

The ramp up here is way too steep for something that's available in such high amounts.

I do not think we should be using "well, it's not as deadly as prions!" as our baseline.

File: code/modules/fishing/fish.dm (hunk 1 above)
Comment (colossusqw) [reply code/modules/fishing/fish.dm:334]: Oversight on account of me looking at rose code, will change.

**Comment by colossusqw** [comment]
Welp. Made overall more obvious early on via making the second to last stage last 20 cycles instead of 5, adding drooling and twitching, as well as custom messages to the early stage, which also starts 2 cycles earlier, making the final stage be 2 cycles later and without paralysis.
For the "it's not as bad as prions!" as a baseline, no, it's not what i mean at all, what i mean is, "this is reasonably deadly for something you can only get from fishing, that doesn't stack well with any other chems, is easily curable, and gives notice that something is wrong".
This absolutely requires you to keep your victim occupied, because they WILL notice something is wrong, and if they get a bit of charcoal in them it's over. It already was mostly a "Columbo poison in the wine RP" poison before, now it's even more so.

**Comment by FronkyGeniuZ** [comment]
Wow really glad for someone work on it appreciate that!

File: code/modules/food_and_drink/snacks.dm [hunk 2]
```diff
@@ -2373,6 +2373,13 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks/soup)
 					nigiri.icon_state = "nigiri3"
 				if("filletslice-small")
 					nigiri.icon_state = "nigiri4"
+				if("filletslice-pufferfish")
+					nigiri.icon_state = "nigiri_pufferfish"
+					nigiri.desc = "A ball of sticky rice with a thin slice of pufferfish fillet ontop. Hopefully properly prepared."
+			// We don't need to be transfering fish oil to nigiri rolls
```
[the remark this answers is not in the document]
Comment (colossusqw) [reply code/modules/food_and_drink/snacks.dm]: Fair

File: code/modules/fishing/fish.dm [hunk 3]
```diff
@@ -285,6 +285,58 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/fish)
...
+				return
+			boutput(H, SPAN_ALERT("YOWCH! You prick yourself on [src]'s spikes! Maybe you should've used gloves..."))
+			random_brute_damage(H, 3)
+			H.setStatusMin("stunned", 2 SECONDS)
+			take_bleeding_damage(H, null, 3, DAMAGE_STAB)
+
+	make_reagents() // No fish oil on this one, more space for poison
+		src.reagents.add_reagent("tetrodotoxin",50) // REALLY don't eat raw pufferfish
+
+	onSlice(var/mob/user) // Don't eat pufferfish the staff assistant made
```
[the remark this answers is not in the document]
Comment (colossusqw) [reply code/modules/fishing/fish.dm:319]: Changed it to transfer to the liver, about to change the transfer to person as well, swapped the 50u and 70u to 40us on both(with a change to 2 slices only so each slice is still lethal)

**Review by Gannets — APPROVED** [review]
@mordent-goonstation says he's happy, I say I'm happy too.

**Pull request merged**